h5py/_hl/files.py

```diff
@@ -348,6 +351,22 @@ def __init__(self, name, mode=None, driver=None,
         track_order
             Track dataset/group/attribute creation order under root group
             if True. If None use global default h5.get_config().track_order.
+        strategy
+            The file space handling strategy to be used.  Only allowed when
+            creating a new file (mode w, w- or x).  Defined as:
```

Comment:
We're having a general push for 'only applicable when' parameters to raise an error if passed in a situation where they're not used, so it would be good to do that for these three.